src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,92 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "ValidLogin";
+    private static final String VALID_PASSWORD = "12345678";
+    private static final int VALID_AGE = 20;
+
+    private static RegistrationServiceImpl regService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        regService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setAge(VALID_AGE);
+        user.setPassword(VALID_PASSWORD);
```

Comment:
You can make it in constructor. And i'm not sure about user in set(if you think that it's normal you can leave it for mentors, because i'm really not sure)

Reply:
There's no constructor tho, so i just used default setters and didn't change the initial class

Reply:
it's bad practice to create User here.
Tests method should be self-explanatory so in such way each method has GIVEN block - initial parameters, WHEN block - which method you are going to test and THEN block - what happened after calling the method.

So it's better to create User in every test method with different parameters for each case.


Also, it's bad to use constants in test. What if you have 100 valid login strings or password? In addition, if you have 50 test methods and more constant other developer should always stick to the beginning of test class to understand what exactly you are using for test. 


So, it's better to give all information what you are testing with particular parameters in each test methods.